document/js-api/index.bs

```diff
@@ -320,12 +330,17 @@ A {{Module}} object represents a single WebAssembly module. Each {{Module}} obje
         1. If [=Type=](|o|) is not Object, throw a {{TypeError}} exception.
         1. Let |v| be [=?=] [$Get$](|o|, |componentName|).
         1. If |externtype| is of the form [=func=] |functype|,
-            1. If [$IsCallable$](|v|) is false, throw a {{LinkError}} exception.
-            1. If |v| has a \[[FunctionAddress]] internal slot, and therefore is an [=Exported Function=],
-                1. Let |funcaddr| be the value of |v|'s \[[FunctionAddress]] internal slot.
-            1. Otherwise,
-                1. [=Create a host function=] from |v| and |functype|, and let |funcaddr| be the result.
-                1. Let |index| be the number of external functions in |imports|. This value |index| is known as the <dfn>index of the host function</dfn> |funcaddr|.
+            1. If [$IsCallable$](|v|) is true
+                1. If |v| has a \[[FunctionAddress]] internal slot, and therefore is an [=Exported Function=],
+                    1. Let |funcaddr| be the value of |v|'s \[[FunctionAddress]] internal slot.
+                1. Otherwise,
+                    1. [=Create a host function=] from |v| and |functype|, and let |funcaddr| be the result.
+            1. If |v| has a \[[wrappedFunction]] insternal slot
+                    1. Let |func| be the value of |v|'s \[[wrappedFunction]] internal slot.
+                    1. Assert: [$IsCallable$](|func|) is true
+                    1. [=Create a suspending function|create a suspending function=] from |func| and |functype|, and let |funcaddr| be the result.
+            1. If [$IsCallable$](|v|) is false or |v| does not have a \[[wrappedFunction]] internal slot throw a {{LinkError}} exception.
+            1. Let |index| be the number of external functions in |imports|. This value |index| is known as the <dfn>index of the host function</dfn> |funcaddr|.
```

Comment:
It is used in line 948: https://github.com/WebAssembly/js-promise-integration/blob/9475b6e13f0687af8d628e2e3fcc7c80c0d4a095/document/js-api/index.bs#L948

Reply:
If I'm not mistaken, that's a definition of a different variable called `|index|` in a different algorithm and is not related to this line here.

Reply:
Perhaps. However:
1. I have not modified this particular text: it was in previous versions of the api spec.
2. What appears to be actually happening is "index of the host function" is being defined. In fact, that is what I was referring to in my previous response (not the use of |index|)

Reply:
> 1. I have not modified this particular text: it was in previous versions of the api spec.

Agreed. Removing this could be done in a separate fixup PR upstream.